test/e2e/deployment.go

```diff
@@ -555,40 +559,117 @@ func (d *Deployment) DeleteResourcesForLocalContour() error {
-	configFile, err := ioutil.TempFile("", "contour-config-*.yaml")
-	if err != nil {
-		return nil, "", err
-	}
-	defer configFile.Close()
+func (d *Deployment) StartLocalContour(config *config.Parameters, contourConfiguration *contour_api_v1alpha1.ContourConfiguration, additionalArgs ...string) (*gexec.Session, string, error) {
+
+	var content []byte
+	var configReferenceName string
+	var contourServeArgs []string
+	var err error
+
+	// Look for the ENV variable to tell if this test run should use
+	// the ContourConfiguration file or the ContourConfiguration CRD.
+	if _, useContourConfiguration := os.LookupEnv("USE_CONTOUR_CONFIGURATION_CRD"); useContourConfiguration {
+		port, _ := strconv.Atoi(d.localContourPort)
+
+		contourConfiguration.Name = randomString(14)
+
+		// Set the xds server to the defined testing port as well as enable insecure communication.
+		contourConfiguration.Spec.XDSServer = contour_api_v1alpha1.XDSServerConfig{
+			Type:    contour_api_v1alpha1.ContourServerType,
+			Address: "0.0.0.0",
+			Port:    port,
+			TLS: &contour_api_v1alpha1.TLS{
+				Insecure: true,
+			},
+		}
 
-	content, err := yaml.Marshal(config)
-	if err != nil {
-		return nil, "", err
-	}
-	if err := os.WriteFile(configFile.Name(), content, 0600); err != nil {
-		return nil, "", err
+		// Disable leader election.
+		contourConfiguration.Spec.LeaderElection = contour_api_v1alpha1.LeaderElectionConfig{
+			DisableLeaderElection: true,
+		}
+
+		if err := d.client.Create(context.TODO(), contourConfiguration); err != nil {
+			return nil, "", fmt.Errorf("could not create ContourConfiguration: %v", err)
+		}
+
+		contourServeArgs = append([]string{
+			"serve",
+			"--kubeconfig=" + d.kubeConfig,
+			"--contour-config-name=" + contourConfiguration.Name,
+		}, additionalArgs...)
+
+		configReferenceName = contourConfiguration.Name
+	} else {
+
+		configFile, err := ioutil.TempFile("", "contour-config-*.yaml")
+		if err != nil {
+			return nil, "", err
+		}
+		defer configFile.Close()
+
+		content, err = yaml.Marshal(config)
+		if err != nil {
+			return nil, "", err
+		}
+		if err := os.WriteFile(configFile.Name(), content, 0600); err != nil {
+			return nil, "", err
+		}
+
+		contourServeArgs = append([]string{
+			"serve",
+			"--xds-address=0.0.0.0",
+			"--xds-port=" + d.localContourPort,
+			"--insecure",
+			"--kubeconfig=" + d.kubeConfig,
+			"--config-path=" + configFile.Name(),
+			"--disable-leader-election",
+		}, additionalArgs...)
+
+		configReferenceName = configFile.Name()
 	}
 
-	contourServeArgs := append([]string{
-		"serve",
-		"--xds-address=0.0.0.0",
-		"--xds-port=" + d.localContourPort,
-		"--insecure",
-		"--kubeconfig=" + d.kubeConfig,
-		"--config-path=" + configFile.Name(),
-		"--disable-leader-election",
-	}, additionalArgs...)
+	// Set the CONTOUR_NAMESPACE env variable so Contour can lookup the Configuration CRD in the proper namespace.
+	//os.Setenv("CONTOUR_NAMESPACE", "projectcontour")
```

Comment:
not needed?

Reply:
I ended up adding this as a default in serve.go, so no not needed.